core/serializers.py

```diff
@@ -80,20 +80,43 @@ def join_team(self):
 
 class HackathonSerializer(serializers.ModelSerializer):
     status = serializers.CharField()
+    userStatus = serializers.SerializerMethodField('get_userStatus')
 
     def validate_end(self, end):
         if end < timezone.now():
             raise serializers.ValidationError('End date cannot be in past')
         return end
+
     def validate_start(self, start):
         if start < timezone.now():
             raise serializers.ValidationError('Start date cannot be in past')
         return start
+
     def validate(self, attrs):
         if attrs['start'] > attrs['end']:
             raise serializers.ValidationError('Event ends before starting')
         return attrs
 
+    def get_userStatus(self, obj):
+        '''
+        Serializer Field to show about the status of the hackathon
+        with respect to the current user
+        '''
+        request = self.context['request']
+        if not request.auth:
+            return 'none'
+        teams = Team.objects.filter(members=request.user, hackathon=obj)
```

Comment:
does directly using = with manytomany field filter works? coz I've seen something like members_in=[request.user] being used

Reply:
Indeed it works. The above filter query produces the following SQL query, which does all the work.
```SQL
SELECT team 
FROM "core_team" INNER JOIN "core_team_members" 
ON ("core_team"."id" = "core_team_members"."team_id") 
WHERE (
    "core_team"."hackathon_id" = ***hackathon_ID*** 
    AND 
    "core_team_members"."user_id" = **********
)
```
here `*****` refers to the userId and `***hackathon_ID***` is the hackathon id

Reply:
ohhhhh